File: campus-server/src/main/java/com/campus/common/result/ApiResult.java
package com.campus.common.result;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ApiResult<T> {

    private int code;
    private String message;
    private T data;

    public static <T> ApiResult<T> success(T data) {
        return new ApiResult<>(0, "success", data);
    }

    public static <T> ApiResult<T> success(String message, T data) {
        return new ApiResult<>(0, message, data);
    }

    public static <T> ApiResult<T> fail(int code, String message) {
        return new ApiResult<>(code, message, null);
    }
}

File: campus-android/app/src/main/java/com/campus/android/core/network/ApiResult.kt
package com.campus.android.core.network

import com.google.gson.annotations.SerializedName

data class ApiResult<T>(
    @SerializedName("code") val code: Int,
    @SerializedName("message") val message: String?,
    @SerializedName("data") val data: T?
)

class CampusApiException(
    val code: Int,
    override val message: String
) : RuntimeException(message)
